src/NServiceBus.Core/StartableEndpoint.cs

```diff
@@ -108,18 +108,33 @@ IEnumerable<TransportReceiver> BuildPipelines(IPipelineCache cache)
             var mainPipelineInstance = new Pipeline<ITransportReceiveContext>(builder, settings, pipelineConfiguration.Modifications);
 
             var pushSettings = new PushSettings(settings.LocalAddress(), errorQueue, purgeOnStartup, requiredTransactionSupport);
-            yield return new TransportReceiver(MainPipelineId, builder, pushSettings, dequeueLimitations, (b, context) => InvokePipeline(b, cache, mainPipelineInstance, context));
+            yield return new TransportReceiver(MainPipelineId,
+                builder,
+                pushSettings,
+                dequeueLimitations,
+                (b, context) => InvokePipeline(b, cache, mainPipelineInstance, context),
+                InvokeError);
 
             if (settings.InstanceSpecificQueue() != null)
             {
                 var sharedReceiverPushSettings = new PushSettings(settings.InstanceSpecificQueue(), errorQueue, purgeOnStartup, requiredTransactionSupport);
-                yield return new TransportReceiver(MainPipelineId, builder, sharedReceiverPushSettings, dequeueLimitations, (b, context) => InvokePipeline(b, cache, mainPipelineInstance, context));
+                yield return new TransportReceiver(MainPipelineId,
+                    builder,
+                    sharedReceiverPushSettings,
+                    dequeueLimitations,
+                    (b, context) => InvokePipeline(b, cache, mainPipelineInstance, context),
+                    InvokeError);
             }
 
             foreach (var satellitePipeline in settings.Get<SatelliteDefinitions>().Definitions)
             {
                 var satellitePushSettings = new PushSettings(satellitePipeline.ReceiveAddress, errorQueue, purgeOnStartup, satellitePipeline.RequiredTransportTransactionMode);
-                yield return new TransportReceiver(satellitePipeline.Name, builder, satellitePushSettings, dequeueLimitations, satellitePipeline.OnMessage);
+                yield return new TransportReceiver(satellitePipeline.Name,
+                    builder,
+                    satellitePushSettings,
+                    dequeueLimitations,
+                    satellitePipeline.OnMessage,
+                    (b, context) => InvokeSatelliteError(b, context));
```

Comment:
should that also be just `InvokeSatelliteError`? If I remember correctly from one of daniel's tests, the normal lambda with the method invocation inside was faster though?